src/mame/shared/tecmo_spr.cpp

```diff
@@ -66,46 +69,48 @@ static const uint8_t layout[8][8] =
  *    5,6,7|                  | unused
  */
 
-#define NUM_SPRITES 256
+static constexpr int NUM_SPRITES = 256;
 
-void tecmo_spr_device::gaiden_draw_sprites(screen_device &screen, gfx_element *gfx, const rectangle &cliprect, uint16_t* spriteram, int sprite_sizey, int spr_offset_y, int flip_screen, bitmap_ind16 &sprite_bitmap)
+void tecmo_spr_device::gaiden_draw_sprites(screen_device &screen, const rectangle &cliprect, uint16_t* spriteram, int sprite_sizey, int spr_offset_y, bool flip_screen, bitmap_ind16 &sprite_bitmap)
 {
-	uint16_t *source = spriteram;
-	int sourceinc = 8;
+	uint16_t const *source = spriteram;
+	int const sourceinc = 8;
 
 	int count = NUM_SPRITES;
-	int screenwidth = screen.visible_area().width();
+	int const screenwidth = screen.visible_area().width();
 
-	int attributes_word = 0;
-	int tilenumber_word = 1;
-	int colour_word = 2;
-	int yposition_word = 3;
-	int xposition_word = 4;
+	int const attributes_word = 0;
+	int const tilenumber_word = 1;
+	int const colour_word = 2;
+	int const yposition_word = 3;
+	int const xposition_word = 4;
 
 	int xmask;
 
-	if (screenwidth == 512)
+	if (screenwidth >= 512)
 		xmask = 512;
 	else
 		xmask = 256;
 
-	/* draw all sprites from front to back */
+	bitmap_ind16* bitmap = &sprite_bitmap;
+
+	// draw all sprites from front to back
 	while (count--)
 	{
-		uint32_t attributes = source[attributes_word];
+		uint16_t const attributes = source[attributes_word];
 
-		int enabled = source[attributes_word] & 0x04;
+		bool enabled = BIT(attributes, 2);
 
 		if (enabled)
 		{
 			if (m_bootleg)
 			{
 				// I don't think the galspinbl / hotpinbl bootlegs have blending, instead they use this bit to flicker sprites on/off each frame, so handle it here (we can't handle it in the mixing)
 				// alternatively these sprites could just be disabled like the tiles marked with the 'mix' bit appear to be (they're only used for ball / flipper trails afaik)
-				if (source[attributes_word] & 0x0040)
+				if (BIT(attributes, 6))
 				{
-					int frame = screen.frame_number() & 1;
-					if (frame==1)
+					uint8_t const frame = screen.frame_number() & 1;
+					if (frame == 1)
 						enabled = 0;
```

Comment:
Since you’ve changed `enabled` to `bool` to reflect usage, you should probably change this to `enabled = false;` to match.